src/arrays/programs_involving_multidimensional_data/easy/q1/AllSameSum.java

```diff
@@ -0,0 +1,33 @@
+package arrays.programs_involving_multidimensional_data.easy.q1;
+
+/*
+A magic square is a box of numbers where each row and column has the same sum. Write a Java program that initialises a
+4 by 4 multidimensional array of integers with values given by the user, and prints true if all the rows have the same
+sum and false otherwise.
+ */
+
+import java.util.Scanner;
+
+public class AllSameSum {
+    public static void main(String[] args) {
+        Scanner input = new Scanner(System.in);
```

Comment:
This program only checks if each row has the same sum, not if each row and column has the same sum (as the question specifies)

Reply:
So we either need to change this so that it checks both, or to change the question to just check the rows